File: 2015/go/day6/day6p2.go
// package day6 contains the solutions to the Day 6 challenge
// of the 2015 Advent of Code challenge.
package day6

import (
	"../utils"
	"fmt"
	"strconv"
	"strings"
)

func Day6Part2() {

	var grid = make([][]int, 1000)
	for i := range grid {
		grid[i] = make([]int, 1000)
	}

	var input = utils.GetInputFileAsString("day6input.txt")
	var lines = strings.Split(input, "\n")
	for _, line := range lines {
		//EXAMPLE turn on 489,959 through 759,964
		tokens := strings.Split(line, " ")
		isToggle := tokens[0] == "toggle"
		var point1Tokens, point2Tokens []string
		if isToggle {
			point1Tokens = strings.Split(tokens[1], ",")
			point2Tokens = strings.Split(tokens[3], ",")
		} else {
			point1Tokens = strings.Split(tokens[2], ",")
			point2Tokens = strings.Split(tokens[4], ",")
		}
		var coord1 []int
		c1x, err := strconv.Atoi(point1Tokens[0])
		c1y, err := strconv.Atoi(point1Tokens[1])
		if err != nil {
			panic(err)
		}
		coord1 = append(coord1, c1x)
		coord1 = append(coord1, c1y)

		var coord2 []int
		c2x, err := strconv.Atoi(point2Tokens[0])
		c2y, err := strconv.Atoi(point2Tokens[1])
		if err != nil {
			panic(err)
		}
		coord2 = append(coord2, c2x)
		coord2 = append(coord2, c2y)

		xMin := coord1[0]
		xMax := coord2[0]

		yMin := coord1[1]
		yMax := coord2[1]

		for i := xMin; i <= xMax; i++ {
			for j := yMin; j <= yMax; j++ {
				if isToggle {
					grid[i][j] += 2
				} else {
					turnOn := tokens[1] == "on"
					if turnOn {
						grid[i][j] += 1
					} else {
						grid[i][j] -= 1
						if grid[i][j] < 0 {
							grid[i][j] = 0
						}
					}
				}
			}
		}
	}

	brightness := 0
	for _, row := range grid {
		for _, b := range row {
			brightness += b
		}
	}
	fmt.Println(brightness)

}

File: 2015/day6/day6p2.py
import sys

file = open("../inputs/day6input.txt", "r")
input = file.read().split("\n")
grid =[[0 for i in range(0,1000)] for j in range(0,1000)]
directions = []


def performDirection(scrub, turnOn, startCoords, endCoords):
	global grid
	startRow = startCoords[0]
	startCol = startCoords[1]
	endRow = endCoords[0]
	endCol = endCoords[1]
	print(startRow, startCol, endRow, endCol)
	for y in range(startRow, endRow+1):
		for x in range(startCol, endCol+1):
			if scrub == False:
					grid[y][x]+=2
			else:
				if turnOn:
					grid[y][x]+=1
				elif grid[y][x] > 0:
					grid[y][x]-=1


for i in input:
	tokens = i.split(" ")
	startCoords = []
	endCoords = []
	#turn on 774,539 through 778,786
	#toggle 341,780 through 861,813
	scrub = False
	turnOn = False
	print(tokens)
	if tokens[0] == "turn":
		scrub = True
		if tokens[1] == "on":
			turnOn = True
		startCoordsStrings = tokens[2].split(",")
		for string in startCoordsStrings:
			startCoords.append(int(string))
		endCoordsStrings = tokens[4].split(",")
		for string in endCoordsStrings:
			endCoords.append(int(string))
	else:
		startCoordsStrings = tokens[1].split(",")
		for string in startCoordsStrings:
			startCoords.append(int(string))
		endCoordsStrings = tokens[3].split(",")
		for string in endCoordsStrings:
			endCoords.append(int(string))

	performDirection(scrub, turnOn, startCoords, endCoords)


count = 0
for row in grid:
	for light in row:
		count += light

print(count)






